### DISCUSSION TITLE
Could really need these two features

### DISCUSSION THREAD

**Discussion by erox455** [shun126/UE5-DungeonGeneratorDemo#17] [Ideas]
- The ability to manually place doors or gates in sublevels, giving us control over the direction of level generation. For example, consider a two-floor starting room with specified doors on the upper floor, or multiple placed exit doors in sublevels that teleport to the exterior.

- An option to generate the level using only premade sublevels, both with and without hallways separating them.

Overall, generation similar to Lethal Company.

**Comment by shun126** [comment]
Lethal Company interesting!

https://www.youtube.com/live/FkyopSbjb_A?si=ufY82Ja_xkYims_t 
I found about this game by watching the stream.

It is interesting how you connect the outdoors with the indoors. Thanks for sharing.

**Reply by shun126** [reply]
It would be expressive if the door could be connected to the second floor. However, it is not possible to support this because the internal information needs to be extended to realize it.
In the sample, I placed a warp actor at the goal to warp to another level.

Generating a dungeon with only sublevels is the way I most wanted to achieve this.
This feature is also not supported because it would require a major change to the order in which the internal information dungeons are generated.

---
二階にドアをつなげられると表現が豊かになりそうです。しかし実現に内部情報を拡張する必要があるため、対応ができません。
サンプルではゴールにワープアクターを配置して、別のレベルにワープするようにしてみました。

サブレベルのみでダンジョンを生成するのは、私も一番実現したかった方法です。
この機能も内部情報のダンジョンの生成順序を変更する大きな変更が必要なため、対応ができません。
